test_conformance/api/test_spirv_queries.cpp

```diff
@@ -0,0 +1,764 @@
+                            &deviceFp64AtomicCapabilities, nullptr);
+        test_error(error,
+                   "clGetDeviceInfo failed for "
+                   "CL_DEVICE_DOUBLE_FP_ATOMIC_CAPABILITIES_EXT\n");
+    }
+
+    // Required.
+    extendedInstructionSets.push_back("OpenCL.std");
+
+    capabilities.push_back(spv::CapabilityAddresses);
+    capabilities.push_back(spv::CapabilityFloat16Buffer);
+    capabilities.push_back(spv::CapabilityInt16);
+    capabilities.push_back(spv::CapabilityInt8);
+    capabilities.push_back(spv::CapabilityKernel);
+    capabilities.push_back(spv::CapabilityLinkage);
+    capabilities.push_back(spv::CapabilityVector16);
+
+    // Required for FULL_PROFILE devices, or devices supporting
+    // cles_khr_int64.
+    if (gHasLong)
+    {
+        capabilities.push_back(spv::CapabilityInt64);
+    }
+
+    // Required for devices supporting images.
+    if (deviceImageSupport == CL_TRUE)
+    {
+        capabilities.push_back(spv::CapabilityImage1D);
+        capabilities.push_back(spv::CapabilityImageBasic);
+        capabilities.push_back(spv::CapabilityImageBuffer);
+        capabilities.push_back(spv::CapabilityLiteralSampler);
+        capabilities.push_back(spv::CapabilitySampled1D);
+        capabilities.push_back(spv::CapabilitySampledBuffer);
+    }
+
+    // Required for devices supporting SPIR-V 1.6.
+    if (ilVersions.find("SPIR-V_1.6") != std::string::npos)
+    {
+        capabilities.push_back(spv::CapabilityUniformDecoration);
+    }
+
+    // Required for devices supporting read-write images.
+    if (deviceReadWriteImageSupport == CL_TRUE)
+    {
+        capabilities.push_back(spv::CapabilityImageReadWrite);
+    }
+
+    // Required for devices supporting the generic address space.
+    if (deviceGenericAddressSpaceSupport == CL_TRUE)
+    {
+        capabilities.push_back(spv::CapabilityGenericPointer);
+    }
+
+    // Required for devices supporting sub-groups or work-group collective
+    // functions.
+    if (deviceSubGroupsSupport == CL_TRUE
+        || deviceWorkGroupCollectiveFunctionsSupport == CL_TRUE)
+    {
+        capabilities.push_back(spv::CapabilityGroups);
+    }
+
+    // Required for devices supporting pipes.
+    if (devicePipeSupport == CL_TRUE)
+    {
+        capabilities.push_back(spv::CapabilityPipes);
+    }
+
+    // Required for devices supporting device-side enqueue.
+    if (deviceDeviceEnqueueSupport == CL_TRUE)
+    {
+        capabilities.push_back(spv::CapabilityDeviceEnqueue);
+    }
+
+    // Required for OpenCL 2.2 devices.
+    if (version == Version(2, 2))
+    {
+        capabilities.push_back(spv::CapabilityPipeStorage);
+    }
+
+    // Required for OpenCL 2.2, or OpenCL 3.0 devices supporting sub-groups.
+    if (version == Version(2, 2)
+        || (version >= Version(3, 0) && deviceSubGroupsSupport == CL_TRUE))
+    {
+        capabilities.push_back(spv::CapabilitySubgroupDispatch);
+    }
```

Comment:
Shouldn't these requirements only apply to devices that support SPIR-V 1.1?

Reply:
I think I'm going to leave the check for SPIR-V 1.1 as-is for now, but after thinking about this a bit more, shouldn't these capability really be supported whenever a device supports SPIR-V 1.1 **or newer**?  The capabilities were added in SPIR-V 1.1, but until they're removed in some future SPIR-V version, I expect they would be available even if a device only supported, say, SPIR-V 1.2.

This might also affect how we document these capabilities in the SPIR-V environment spec.  Maybe I should file an issue for this, specifically.

Reply:
I'd like to leave this as-is for now, is possible, in the interest of getting these changes in soon (we're already seeing some "failures" due to the missing changes to compiler_defines_for_extensions).  My rationale is:

* This _probably_ isn't an issue in practice currently, since most implementations that advertise support for newer versions of SPIR-V (e.g. 1.2 or newer) also advertise support for the older versions of SPIR-V (1.1)
* Even if an implementation only advertises support for the newer versions of SPIR-V, we'll just skip a few checks, and the tests won't "fail" unexpectedly.

I'd be happy to file an issue to fix this so it doesn't slip through the cracks.